darts/models/forecasting/regression_model.py

```diff
@@ -701,9 +702,22 @@ def _fit_model(
                 val_sample_weight=val_sample_weight,
                 max_samples_per_ts=max_samples_per_ts,
             )
-        self.model.fit(
-            training_samples, training_labels, sample_weight=sample_weights, **kwargs
-        )
+
+        # check whether model's fit method supports parameter sample_weights
+        if sample_weights is not None:
+            fit_parameters = inspect.signature(self.model.fit).parameters
+            if "sample_weight" in fit_parameters:
+                self.model.fit(
+                    training_samples,
+                    training_labels,
+                    sample_weight=sample_weights,
+                    **kwargs,
+                )
+            else:
+                logger.warning("Provided `sample_weights` wasn't used.")
+                self.model.fit(training_samples, training_labels, **kwargs)
+        else:
+            self.model.fit(training_samples, training_labels, **kwargs)
```

Comment:
Just a slight modifictation to write the `fit()` call only once :)

```suggestion
        # check whether model's fit method supports parameter `sample_weight`
        sample_weight_kwargs = dict()
        if sample_weights is not None:
            if "sample_weight" in inspect.signature(self.model.fit).parameters:
                sample_weight_kwargs = {"sample_weight": sample_weights}
            else:
                logger.warning(
                    "`sample_weight` was ignored since underlying regression model's "
                    "`fit()` method does not support it."
                )
        self.model.fit(
            training_samples, training_labels, **sample_weight_kwargs, **kwargs
        )
```

Reply:
Thank you very much for your advice! @dennisbader Of course, I will complete the code as soon as possible and add some describition of this fix in the changelog.

Reply:
Hi! @dennisbader I've completed the code changes and changelog update, please check it out.🤝